mood-analysis.rb

```diff
@@ -19,6 +20,58 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def strip_punctuation(words)
+  words.gsub(/[!.#,]/, "")
+end
+
+def happy_days(array)
+  entries = 0
+  happy_faces = 0
+  array.each do |line|
+    if happy_faces < 3
+      emotion = analyze_mood(line)
+      entries += 1
+      if emotion == ":-)"
+      happy_faces += 1
+      break if happy_faces == 3
+      end
+    end
+  end
+  if happy_faces == 3
+    return "It takes #{entries} entries for 3 happy days to occur"
+  else
+    return "Not enough happy days!"
+  end
+end
+
+def overall_mood(array)
+  happy_moods = []
+  sad_moods = []
+  neutral_moods =[]
+  array.each do |line|
+    moods = analyze_mood(line)
+    if moods == ":-)"
+      happy_moods << moods
+    elsif moods == ":-("
+      sad_moods << moods
+    else
+      neutral_moods << moods
+    end
+  end
+  happy_length = happy_moods.length
+  sad_length = sad_moods.length
+  neutral_length = neutral_moods.length
+
+  if happy_length > sad_length && happy_length > neutral_length
+    return "The most common mood is :-)"
+  elsif sad_length > happy_length && sad_length > neutral_length
+    return "The most common mood is :-("
+  else
+    return "The most common mood is :-|"
+  end
+end
```

Comment:
This works, but a hash would have made for less work and a more scalable solution.